app/models/dependency.rb

```diff
@@ -2,6 +2,7 @@ class Dependency < ApplicationRecord
   include DependencyChecks
 
   belongs_to :version
+  belongs_to :version_project, class_name: 'Project'
```

Comment:
Also considered `dependent_project` as a name? But this is more obvious at first glance.